### PR TITLE
Add sdl2 system cursors

### PR DESCRIPTION
**Pull request by Bakterija**
Hello.
I have tested kivy  with sdl2 system cursors on windows, ubuntu, android 4.4 and everything works very nicely. I would like to share it with other people.

This commit adds necessary structs from SDL2 and a set_system_cursor method for kivy window where you can replace your app cursor.

### CODE REVIEW

**Comment by KeyWeeUsr** [comment]
Seems nice! If no one shows for testing, I'll look into it soon-ish.

**Comment by KeyWeeUsr** [comment]
Everything except `wait_arrow` works just fine. Does it work for you? Share a screenshot, please.

File: kivy/core/window/__init__.py [hunk 1]
```diff
@@ -732,6 +732,19 @@ def _get_focus(self):
     def _set_cursor_state(self, value):
         pass
 
+    def set_system_cursor(self, cursor_name):
+        '''Set cursor to a SDL_SystemCursor
```
Comment (KeyWeeUsr) [inline kivy/core/window/__init__.py]: Run the docstring e.g. through http://rst.ninjs.org and fix the formatting, please :)

**Comment by Bakterija** [comment]
Wait arrow is a crosshair for me. It depends on platform.

File: kivy/core/window/__init__.py (hunk 1 above)
Comment (Bakterija) [inline kivy/core/window/__init__.py]: I am not sure what you want to fix, it looks like other kivy docstrings.

Comment (Bakterija) [inline kivy/core/window/__init__.py]: I am not sure what you want to fix, it looks like other kivy docstrings.

**Comment by KeyWeeUsr** [comment]
Yes, it depends on a platform, but neither of those should return a crosshair. See [windows](https://hg.libsdl.org/SDL/file/c728c661cec7/src/video/windows/SDL_windowsmouse.c#l174), [wayland](https://hg.libsdl.org/SDL/file/c728c661cec7/src/video/wayland/SDL_waylandmouse.c#l261), [cocoa/mac](https://hg.libsdl.org/SDL/file/c728c661cec7/src/video/cocoa/SDL_cocoamouse.m#l133), [x11](https://hg.libsdl.org/SDL/file/c728c661cec7/src/video/x11/SDL_x11mouse.c#l245), [docs](https://wiki.libsdl.org/SDL_CreateSystemCursor) :)

I think you ran an example like this:

```python
from kivy.app import runTouchApp
from kivy.lang import Builder
runTouchApp(Builder.load_string('''
#:import win kivy.core.window.Window

GridLayout:
    cols: 4
    Button:
        text:  'arrow'
        on_release: win.set_system_cursor('arrow')
    Button:
        text:  'ibeam'
        on_release: win.set_system_cursor('ibeam')
    Button:
        text: 'wait'
        on_release: win.set_system_cursor('wait')
    Button:
        text: 'crosshair'
        on_release: win.set_system_cursor('crosshair')
    Button:
        text: 'wait_arrow'
        on_release: win.set_system_cursor('wait_arrow')
    Button:
        text: 'size_nwse'
        on_release: win.set_system_cursor('size_nwse')
    Button:
        text: 'size_nesw'
        on_release: win.set_system_cursor('size_nesw')
    Button:
        text: 'size_we'
        on_release: win.set_system_cursor('size_we')
    Button:
        text: 'size_ns'
        on_release: win.set_system_cursor('size_ns')
    Button:
        text: 'size_all'
        on_release: win.set_system_cursor('size_all')
    Button:
        text: 'hand'
        on_release: win.set_system_cursor('hand')
    Button:
        text: 'no'
        on_release: win.set_system_cursor('no')
'''))
```

and changed the icons one after another. Try to press only the `wait_arrow` button and nothing else and see if it even changes the icon. It doesn't for me on windows and that one uses the [latest stable](https://github.com/kivy/kivy-sdk-packager/blob/master/win/sdl2.py#L9) SDL2, which probably means it's an upstream bug, I guess.

If you have the same result, it might be a good idea to include a note about that.

**Pull request merged**